### PR TITLE
QEMU VM's based providers: run etcd on RAM memory 

### PR DESCRIPTION
**Pull request by ormergi**
On CI and dev environments where the cluster is ephemeral and
data shouldn't be persistent it is possible to improve cluster
performance by configuring etcd to run on RAM memory.
Hopefully it will shorten the time it takes for CI job to
finish and improve CI capacity.

This is a following to what we did for KIND based provider, 
it does shorten the time it takes for a cluster to be ready and 
improve the cluster overall performance.

### CODE REVIEW

**Comment by rmohr** [comment]
@ormergi I was curious on friday on how to do this and we just did this: https://github.com/kubevirt/kubevirtci/pull/498. There is a PR against kubevirt/kubevirt: https://github.com/kubevirt/kubevirt/pull/4662. I want to see  it running on the k8s 1.19 lane for some time. The PR is currently however blocked due to an unrelated issue. There seems to be no immediate improvement, but I want to see it running for some time and see if it is helpful when ci nodes are a little bit more under pressure.

**Comment by ormergi** [comment]
> @ormergi I was curious on friday on how to do this and we just did this: #498. There is a PR against kubevirt/kubevirt: [kubevirt/kubevirt#4662](https://github.com/kubevirt/kubevirt/pull/4662). I want to see it running on the k8s 1.19 lane for some time. The PR is currently however blocked due to an unrelated issue. There seems to be no immediate improvement, but I want to see it running for some time and see if it is helpful when ci nodes are a little bit more under pressure.

Well I think we cant see immediate significant improvement because QEMU providers are pretty much optimized
we never encountered etcd issues with it before, 
I guess it because it always run on CI nodes that has fast disks, and  If for any reason it would run on a node with slow disk you would see immediate improve.
We can start to test the limmits of how many jobs could run on a single node with etcd on memory ;)

File: cluster-provision/gocli/cmd/run.go [hunk 1]
```diff
@@ -83,6 +87,7 @@ func NewRunCommand() *cobra.Command {
 	run.Flags().String("container-org", "kubevirtci", "the organization at the registry to pull the container from")
 	run.Flags().String("container-suffix", "", "Override container suffix stored at the cli binary")
 	run.Flags().String("gpu", "", "pci address of a GPU to assign to a node")
+	run.Flags().Bool("run-etcd-on-memory", true, "configure etcd to run on RAM memory, etcd data will not be persistent")
```
Comment (rmohr) [inline cluster-provision/gocli/cmd/run.go:90]: Can you make this an it to specify MBs? The boolean is nice, but I saw in #498 that we may have to vary the size. Maybe just add another flag for that.

**Review by rmohr — COMMENTED** [review]
I am currently reverting #498 and would want to get your PR in. One suggestion regarding to making the ramdisk size configurable.

File: cluster-provision/gocli/cmd/run.go (hunk 1 above)
Comment (ormergi) [reply cluster-provision/gocli/cmd/run.go:90]: No problem

**Comment by ormergi** [comment]
Changes:
- Added flag to make RAM disk mount size for etcd data configurable
- Add validation to data size (e.g: 200K, 500M, 1g) parameters

**Comment by ormergi** [comment]
Fixed typos on commit messages, had to force pushed

**Comment by ormergi** [comment]
Changes:
- Added new env variable to control etcd data mount size
- Adapted ￼ ephemeral-provider-common.sh
- re-arranged commit order

**Comment by ormergi** [comment]
> Seems like we have a compile issue:
> 
> ```
> vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go:470:34: invalid operation: int64(1) << (exponent * 10) (shift count type int, must be unsigned integer)
> ```

@rmohr 
I noticed that on gocli makefile we use the host `go` binary, maybe this is the problem
since my local setup I dont get this error.
Do you know which go version we use in check-provision jobs?

**Comment by ormergi** [comment]
Changes: 
- Pin k8s.io/apimachinery package version to v0.18.4

**Comment by rmohr** [comment]
This sounds like an easy fix:
```
flag accessed but not defined: etcd-capcity
```

**Comment by ormergi** [comment]
> This sounds like an easy fix:
> 
> ```
> flag accessed but not defined: etcd-capcity
> ```

Thanks for the heads up, I rebased and fixed it

**Comment by ormergi** [comment]
@rmohr CI is green

**Pull request merged**